hls4ml/converters/onnx/pooling.py

```diff
@@ -0,0 +1,58 @@
+import math
+from hls4ml.converters.onnx_to_hls import onnx_handler, get_onnx_attribute, compute_pads_1d, compute_pads_2d
+
+pool_operations = ['AveragePool', 'MaxPool']
+@onnx_handler(*pool_operations)
+def parse_pool_layer(reader, node, inputs_map, input_shapes, graph, config):
+            
+    info = layer['class_name'].replace('Pool', '')
+    strides = get_onnx_attribute(node, 'strides')
+    kernel_shape = get_onnx_attribute(node, 'kernel_shape')
+    
+    if len(current_shape) == 3: # 1D
+        layer['class_name'] = info + 'Pooling1D'
+        layer['stride'] = strides[0]
+        layer['pool_size'] = layer['y_filt'] = kernel_shape[0]
+        
+        #Padding
+        pads = compute_pads_1d(node, layer)
+        layer['pad_left'] = pads[0]
+        layer['pad_right'] = pads[1]
+
+        if all(x == 0 for x in pads): # No padding, i.e., 'VALID' padding
+            layer['n_out'] = int(math.ceil(float(layer['y_in'] - layer['y_filt'] + 1) / float(layer['stride'])))
```

Comment:
Same as with convolution, computing output shape should be shared by all parsers